smolyay/samples.py

```diff
@@ -508,3 +511,583 @@ def _create(self):
+    optional parameters, ``scramble`` and ``optimization``.
+
+    Parameters
+    ----------
+    domain: list
+        domain of the random points
+
+    num_points: int
+        number of points to generate
+
+    seed: int
+        seed for generating the random points
+
+    scramble : bool
+        Default True. Applies centering to Latin Hypercube points, Owen
+        scrambling to Halton points, and LMS+shift scrambling to Sobol points.
+
+    optimization : {None, "random-cd", "lloyd"}
+        Default None. If "random-cd" the coordinates of points are adjusted to
+        lower the centered discrepancy. If "lloyd", adjust points using a
+        Lloyd-Max algorithm to encourage even spacing.
+    """
+
+    def __init__(self, domain, num_points, seed, scramble=True, optimization=None):
+        super().__init__(domain, num_points, seed)
+        self._scramble = True
+        self._optimization = None
+
+        self.scramble = scramble
+        self.optimization = optimization
+
+    @property
+    def scramble(self):
+        """bool: If True, applies centering to Latin Hypercube points, Owen
+        scrambling to Halton points, and LMS+shift scrambling to Sobol points."""
+        return self._scramble
+
+    @scramble.setter
+    def scramble(self, value):
+        scramble = bool(value)
+        if self._scramble != scramble:
+            self._scramble = scramble
+            self._scramble = False
+
+    @property
+    def optimization(self):
+        """{None, “random-cd”, “lloyd”}: perform post-processing on points"""
+        return self._optimization
+
+    @optimization.setter
+    def optimization(self, value):
+        if not value in [None, "random-cd", "lloyd"]:
+            raise TypeError("optimization must be None, random-cd, or lloyd")
+        if self._optimization != value:
+            self._optimization = value
+            self._valid_cache = False
+
+
+class LatinHypercubeRandomPointSet(_QMCRandomPointSet):
+    """Generates a grid using a LatinHypercube
+
+
+    Parameters
+    ----------
+    domain: list
+        domain of the random points
+
+    num_points: int
+        number of points to generate
+
+    seed: int
+        seed for generating the random points
+
+    scramble : bool
+        Default True. Applies centering to Latin Hypercube points.
+
+    optimization : {None, "random-cd", "lloyd"}
+        Default None. If "random-cd" the coordinates of points are adjusted to
+        lower the centered discrepancy. If "lloyd", adjust points using a
+        Lloyd-Max algorithm to encourage even spacing.
+
+    strength : {1, 2}, optional
+        Default 1. If 1, produces a normal LHS. If 2, produces an orthogonal
+        array based LHS.
+    """
+
+    def __init__(
+        self, domain, num_points, seed, scramble=True, optimization=None, strength=1
+    ):
+        super().__init__(domain, num_points, seed, scramble, optimization)
+        self._strength = 1
```

Comment:
Is there a reason not to assign this as None initially? The behavior is slightly an anti-pattern.
```suggestion
        self._strength = None
```